src/app/public/modules/href/href-resolver.ts

```diff
@@ -0,0 +1,5 @@
+import { AppRoute } from './types/app-route';
+
+export interface SkyHrefResolver {
+  resolveHref(url: string): Promise<AppRoute>;
```

Comment:
It's possible that `resolveHref` will be extended with new optional parameters, in which case I prefer to have one `args` parameter where new properties can added later without cluttering up the method signature. I would `SkyHrefResolverArgs` interface to represent the shape of that `args` parameter and make `url` one of its required properties.